src/components/ResortsData.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react';
+import ResortsList from './ResortsList';
+import ResortsListHead from './ResortsListHead';
+import Pagination from './Pagination';
+
+class ResortsData extends Component{
+  constructor(){
+    super();
+    this.state = {
+      data: [],
+      page: 3,
+      isLoading: false,
+      error: null
+    }
+  }
+  
+  fetchResorts(){
+    this.setState({isLoading: true})
+    fetch(`http://localhost:3000/resorts?_page=${this.state.page}&_limit=1`)
+        .then(response => {
+          if (response.ok) {
+            return response.json();
+          } else {
+            throw new Error('Something went wrong ...');
+          }
+        })
+        .then(data => this.setState({ data: data, isLoading: false }))
+        .catch(error => this.setState({ error, isLoading: false}))
+  }
+
+  componentDidMount(){
+    this.fetchResorts();
+  }
+
+  handleClick(page) {
+    this.setState({page: page},() => this.fetchResorts() );
```

Comment:
Other ways of doing this are:
- Call fetchResorts in `componentWillUpdate` lifecycle hook and trigger it when page state changes.
- Remove `page` from state and in `handleClick` just call `fetchResorts(page)` (it would require fetchResorts to accept argument and use it instead of state)

Which one would you prefer and why?

Reply:
I think that the second one is better because actually page doasn't work as a component state - we dont use it to change component state directly, it is just way to indicate from where data have to be fetch in fetch method.
Is it OK?

Reply:
In the current state of the application it is sufficient, but keep in mind, that soon we'll need to implement full pagination, so we need to know which page is currently active. (as well as other data required for pagination to work correctly)

Reply:
OK I'll implement the first one :)

Reply:
tomorrow...